pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -259,16 +267,17 @@ func (a *index) constructService(ctx krt.HandlerContext, svc *v1.Service, w *Way
 			ipFamily = workloadapi.IPFamilies_IPV6_ONLY
 		}
 	}
-	// TODO this is only checking one controller - we may be missing service vips for instances in another cluster
+	// TODO: this is only checking one controller - we may be missing service vips for instances in another cluster
 	return &workloadapi.Service{
-		Name:          svc.Name,
-		Namespace:     svc.Namespace,
-		Hostname:      string(kube.ServiceHostname(svc.Name, svc.Namespace, a.DomainSuffix)),
-		Addresses:     addresses,
-		Ports:         ports,
-		Waypoint:      w.GetAddress(),
-		LoadBalancing: lb,
-		IpFamilies:    ipFamily,
+		Name:                svc.Name,
+		Namespace:           svc.Namespace,
+		Hostname:            string(kube.ServiceHostname(svc.Name, svc.Namespace, a.DomainSuffix)),
+		Addresses:           addresses,
+		Ports:               ports,
+		Waypoint:            w.GetAddress(),
+		LocalNetworkGateway: networkGateway.GetAddress(),
```

Comment:
Aside from debate in the design doc about whether this approach is right at a high level -- do we need this in the workloadapi.Service or just the model.ServiceInfo? Seems we only use it internally within Istiod

Reply:
Hmm good point. Yes it's only used in istiod so model.ServiceInfo may be better. Need to check what the krt impl would look like